softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -145,9 +145,32 @@ func resourceSoftLayerVirtualGuest() *schema.Resource {
 			},
 
 			"dedicated_acct_host_only": {
-				Type:     schema.TypeBool,
-				Optional: true,
-				ForceNew: true,
+				Type:          schema.TypeBool,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"dedicated_host_name", "dedicated_host_id"},
+			},
+
+			"dedicated_host_name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				ConflictsWith: []string{"dedicated_acct_host_only", "dedicated_host_id"},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					_, ok := d.GetOk("dedicated_host_id")
+					return new == "" && ok
+				},
```

Comment:
Is this func necessary even with the conflictsWith?

Reply:
Yes it is needed. Because when you create VM terraforms saves `dedicated_host_name` and `dedicated_host_id` to `tfstate`. Then if you apply it again it gets empty string for `dedicated_host_name` from `*.tf` file and actual name from `tfstate`.
Without this function terraform recreates VM every time.